toolchain/semantics/semantics_ir.h

```diff
@@ -59,17 +62,26 @@ class SemanticsIR {
  private:
   friend class SemanticsParseTreeHandler;
 
-  // As noted under cross_reference_irs_, the current IR must always be at
-  // index 1. This is a constant for that.
-  static constexpr auto ThisIR = SemanticsCrossReferenceIRId(1);
-
   // For the builtin IR only.
   SemanticsIR() : SemanticsIR(*this) {}
   // For most IRs.
   SemanticsIR(const SemanticsIR& builtins)
       : cross_reference_irs_({&builtins, this}),
         cross_references_(builtins.cross_references_) {}
 
+  // Returns the requested node, resolving cross references.
+  auto GetNode(SemanticsNodeBlockId block_id, SemanticsNodeId node_id)
+      -> SemanticsNode {
```

Comment:
There's a subtlety here: the node IDs in the returned `SemanticsNode` might refer to nodes in a different IR or a different block of the same IR, so the caller won't be able to interpret those references. It seems like in order to handle that properly we might need to return something more like a `SemanticsNodeRef` that carries an (IR index, block index, `SemanticsNode`) triple.

Reply:
Just to quickly note, since I'm not sure I'll fix the Emit note right now, I think #2475 will address most of this. There may still be a need to deal with cross-IR references, but it eliminates the block problem, as well as cross_references_ itself (which is part of the complexity).